### DISCUSSION TITLE
Is there a way to make the parser forward-only?

### DISCUSSION THREAD

**Discussion by Niedzwiedzw** [jam1garner/binrw#304] [Q&A]
I am parsing from a reader which doesn't implement Seek. I have implemented a ForwardOnlySeek<R> wrapper which errors on backwards seek but looks like the parser sometimes goes back, I think it's when matching enum variants... is there a way around it?

**Comment by csnover** [comment]
Hi, thanks for your question! binrw already [comes with a wrapper](https://docs.rs/binrw/latest/binrw/io/struct.NoSeek.html) for this, so you don’t need to write your own. Issue #253 tracks optimisations to avoid re-reading magic on data enums, and a workaround is given in the comments of that ticket.
